src/Builder/ArchiveBuilder.php

```diff
@@ -119,7 +119,8 @@ public function dump(array $packages): void
                 $distUrl = sprintf('%s/%s/%s/%s', $endpoint, $this->config['archive']['directory'], $intermediatePath, $archive);
                 $package->setDistType($archiveFormat);
                 $package->setDistUrl($distUrl);
-                $package->setDistSha1Checksum($includeArchiveChecksum ? hash_file('sha1', $path) : null);
+                $hasedPath = $includeArchiveChecksum ? hash_file('sha1', $path) : false;
+                $package->setDistSha1Checksum(false !== $hasedPath ? $hasedPath : null);
```

Comment:
`hash_file` will never return a falsy string (`''` or `'0'`)
```suggestion
                $package->setDistSha1Checksum($includeArchiveChecksum ? (hash_file('sha1', $path) ?: null) : null);
```

Reply:
The return type for `hash_file` is `string|false` and this rule level disallows short ternary operators, I'll tweak this a little though.